tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/macro/MacroGeneratorIntegrationTest.xtend

```diff
@@ -0,0 +1,282 @@
+package org.testeditor.tcl.dsl.jvmmodel.macro
+
+import org.junit.Before
+import org.junit.Test
+import org.testeditor.dsl.common.testing.DummyFixture
+
+class MacroGeneratorIntegrationTest extends org.testeditor.tcl.dsl.jvmmodel.AbstractTclGeneratorIntegrationTest {
+
+	@Before
+	def void setup() {
+		parseAml(DummyFixture.amlModel)
+		parseTcl('''
+			package com.example
+			
+			# MyMacroCollection
+			
+			## EmptyMacro
+				template = "Do nothing"
+			
+			## EmptyNestedMacro
+				template = "Do nothing nested"
+				Macro: MyMacroCollection
+				- Do nothing
+			
+			## EmptyMacroWithUnusedParameter
+				template = "Do nothing with" ${unused}
+			
+			## ReadMacro
+				template = "Read some values"
+				Component: GreetingApplication
+				- value = Read value from <bar>
```

Comment:
Yes it might be a warning for the validator. But adding usage step here does not improve the test case - which is if a repeated invocation can be done (which was not the case before).